src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java

```diff
@@ -320,9 +310,12 @@ public static int upload(Run<?, ?> run, Launcher launcher, TaskListener listener
 			// Remove the workspace bit of this path
 			URI srcURI = workspaceURI.relativize(src.toURI());
 
-			InputStream inputStream = src.read();
-			String md5hex = DigestUtils.md5Hex(inputStream);
-			long sizeInBytes = src.length();
+			String md5hex;
+			long sizeInBytes;
+			try (InputStream inputStream = src.read()){
+					md5hex = DigestUtils.md5Hex(inputStream);
```

Comment:
If that doesn't work out, you should at least wrap the same md5hex logic below for the 'archiveFile' in a try block